lib/buckets.ps1

```diff
@@ -58,10 +58,18 @@ function Get-LocalBucket {
     .SYNOPSIS
         List all local buckets.
     #>
-    $bucketNames = (Get-ChildItem -Path $bucketsdir -Directory).Name
+    $bucketNames = [System.Collections.Generic.List[String]](Get-ChildItem -Path $bucketsdir -Directory).Name
     if ($null -eq $bucketNames) {
         return @() # Return a zero-length list instead of $null.
     } else {
+        $knownBuckets = known_buckets
+        for ($i = $knownBuckets.Count - 1; $i -ge 0 ; $i--) {
+            $name = $knownBuckets[$i]
+            if ($bucketNames.Contains($name)) {
```

Comment:
This method will sort names that are same as official bucket names, but still have a different/custom URL. We should check for bucket URL as well while sorting.

Reply:
I've thought of it, but checking URL is too time-consuming and one solution is only topping `main`, `extras` and `version` buckets (or adding `non-portable`) to avoid conflict.

This solves most problems, and someone could use the official buckets' name as their own buckets at their own risks.

Reply:
@niheaven Agreed. And we don't need to check URLs because someone may use a proxy/mirror URL for known buckets, e.g., `https://ghproxy.com/https://github.com/ScoopInstaller/Main` for the `main` bucket, which is what I use.

Reply:
Hmm okay